src/compiler/types.rs

```diff
@@ -203,6 +302,8 @@ pub struct BaseNode {
     pub parent: RwLock<Option<Weak<Node>>>,
     pub pos: AtomicUsize,
     pub end: AtomicUsize,
+    pub symbol: RefCell<Option<Weak<Symbol>>>,
```

Comment:
I started wondering why I was using `RwLock` instead of `RefCell` (since it seems like you see `Rc<RefCell<...>>`'s more than `Rc<RwLock<...>>`'s). I'm not totally sure but I think I can replace a lot of the `RwLock`'s with `RefCell`'s and that per eg [this article](https://manishearth.github.io/blog/2015/05/27/wrapper-types-in-rust-choosing-your-guarantees/) `RefCell`'s sound cheaper/faster

Reply:
Ah ok, yeah I had no idea that `RefCell` was just the single-threaded version of `RWLock`. (The naming doesn't really help there.)